mbed_build/_internal/config/assemble_config.py

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Configuration assembly."""
+import pathlib
+
+from mbed_build._internal.config.config import Config
+
+
+def assemble_config(mbed_target: str, mbed_program_directory: pathlib.Path) -> Config:
+    """Assemble Config for given target and program directory.
+
+    The structure and configuration of MbedOS requires us to do multiple passes over
+    configuration assembly, as each Source can modify filtering rules applied to finding libs.
+    """
+    return Config()
```

Comment:
Is this WIP? I suppose you have another PR ready which passes the arguments through?

Reply:
Yeah, I had this in first, but decided to drop the code as it was untested. Planning to add this as next step. Do you think this file should be removed? 🤔

Reply:
FWIW, rationale was - this allows for a separate thread where a command to display config can be implemented. `assemble_config` is the interface to retrieve it.